### PR TITLE
add schema for helm and dockerlayer assets

### PR DESCRIPTION
**Pull request by dexhorthy**

### CODE REVIEW

File: projects/replicated-ship/schema.json [hunk 1]
```diff
@@ -75,29 +74,98 @@
...
+                  }
+                ],
+                "type": "object",
+                "properties": {
+                  "description": {
                     "type": "string"
                   },
-                  "repo": {
+                  "dest": {
+                    "description": "The directory in which to render the. If the source chart is at `charts/src/nginx`, and `dest` is set to `charts/rendered/`, then the chart will be templated at `charts/rendered/nginx`",
```
Comment (laverya) [inline projects/replicated-ship/schema.json]: `The directory in which to render the.` 

This is probably missing something

File: projects/replicated-ship/schema.json [hunk 2]
```diff
@@ -75,29 +74,98 @@
...
+                },
+                "required": [
+                  "dest",
+                  "image",
+                  "source",
+                  "layer"
+                ]
+              },
+              "helm": {
+                "description": "A `helm` asset will template and render an existing helm chart. Your installer runbook should included assets to drop an existing helm chart at `local.chart_root`. Future releases will include support for charts stored in private github repos or in chart repositories like ChartMuseum.",
```
Comment (laverya) [inline projects/replicated-ship/schema.json:102]: `should included` -> should include

File: projects/replicated-ship/schema.json [hunk 3]
```diff
@@ -75,29 +74,98 @@
...
+                  "image": {
+                    "description": "The docker image URL",
+                    "type": "string"
+                  },
+                  "layer": {
+                    "description": "a SHA256 of the layer to extract. You can inspect the layer SHAs of a docker image by running something like `docker save myimage > myimage.tar && tar xvf myimage.tar -C /tmp`. Ship will unpack the archive at `<layer sha>/layer.tar` to the `dest` directory.",
                     "type": "string"
                   },
                   "mode": {
+                    "description": "The unix file permissions to be set on target unpack directory, e.g `777`",
```
Comment (laverya) [inline projects/replicated-ship/schema.json]: Should this be part of the example? And should we list the default?

File: projects/replicated-ship/schema.json [hunk 4]
```diff
@@ -58,15 +58,14 @@
                   "source"
                 ]
               },
-              "github": {
-                "description": "A `github` asset is created from files downloaded from either a public or a linked Github repo.",
+              "dockerlayer": {
+                "description": "A `dockerlayer` asset will pull an image from registry.replicated.com or from a configured third party private registry and extract a single layer from the image in a `dest` directory",
```
Comment (laverya) [inline projects/replicated-ship/schema.json]: No public source support?

Comment (dexhorthy) [reply projects/replicated-ship/schema.json]: we have public source support

File: projects/replicated-ship/schema.json (hunk 1 above)
Comment (dexhorthy) [reply projects/replicated-ship/schema.json]: i swear I fixed this cmon github https://github.com/replicatedhq/ship/pull/115#discussion_r196141659

**Review by laverya — APPROVED** [review]

**Pull request merged**